src/inmanta/agent/agent.py

```diff
@@ -211,6 +211,13 @@ async def execute(
             # Explicit cast is required because mypy has issues with * and generics
             results: List[ResourceActionResult] = cast(List[ResourceActionResult], await asyncio.gather(*waiters))
 
+            if self.undeployable:
+                self.status = self.undeployable
+                self.change = const.Change.nochange
+                self.changes = {}
+                self.future.set_result(ResourceActionResult(cancel=False))
+                return
```

Comment:
Don't we need to briefly set `self. running` as well to avoid the race condition? Or is there no risk of that because we have no `await` here, therefore no context switching and we're only using a single thread?

Reply:
This is a bug. I implemented the fix.